src/main/java/com/github/argon/sos/moreoptions/game/booster/FactionOpinionBooster.java

```diff
@@ -1,22 +1,24 @@
 package com.github.argon.sos.moreoptions.game.booster;
 
+import game.boosting.BSourceInfo;
 import game.boosting.BoosterImp;
+import game.faction.Faction;
 import game.faction.npc.ruler.Royalty;
 
-public class FactionOpinionBooster extends BoosterImp<Royalty> {
+public class FactionOpinionBooster extends BoosterImp {
 
     private double value = 0.0;
 
-    public FactionOpinionBooster(CharSequence sourceName, double from, double to, boolean isMul) {
-        super(sourceName, from, to, isMul);
+    public FactionOpinionBooster(BSourceInfo bSourceInfo, double from, double to, boolean isMul) {
+        super(bSourceInfo, from, to, isMul);
     }
 
     public void set(double value) {
         this.value = value;
     }
 
     @Override
-    public double get(Royalty royalty) {
-        return value;
+    public double vGet(Faction f) {
+        return 0;
```

Comment:
returning 0 would make the booster useless I guess

Reply:
still WIP, I want to use it maybe, the Opinion boosters are their unique kind